### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by Zod-**
This PR doesn't change any logic in the main Project. I've renamed the `EditorUtils` to `TestUtils` to make it more clear that they are not to be used in the main project since I'm using some reflection in there.

The main functionality of `TestUtils` is the `ClearScene` method. I try to call this in most `TearDown` and `SetUp` methods in each test. The reason behind this is that unity does not delete objects in the test scene after it ran them and some tests rely on there being no previous objects.
For example this test:
```csharp
        [Test]
        public void GetNullCameraFromCache()
        {
            Assert.That(CameraCache.Main == null, Is.True);
        }
```

If some other test created a camera before this one ran it would not return null and make my test return false results. 

Another important functionality is being able to call `Awake`, `Start` and `Update` methods since they will not be called when a script is instantiated in the test scene. 
For example, I create a cube in front of the camera that implements `IFocusable` in the scene and then simulates 1 frame with the `GazeManager` by calling its update and check if the cube received the event. 

```csharp
        [Test]
        public void FocusChangeFullIntegration()
        {
            var handler = CreateCubeTestHandler().CallAwake().CallStart();

            GazeManager.Instance.gameObject.CallUpdate();

            Assert.That(receivedEventSources.Count, Is.EqualTo(1));
            Assert.That(receivedEventSources[0], Is.EqualTo(handler));
        }
```

I've added tests for `CameraCache`, `ActionExtensions`, `EnsureComponent`, almost all cases in `InputManager.HandleEvent` and that one full integration test with the `GazeManager` you see above.

Another thing to keep in mind when using NUnit in combination with Unity is that it comes with the same issues when comparing unity objects to null. `Is.Null` will not return the same result as `gameObject == null`. I've tried making a custom constraint for this but I haven't figured out a good name yet and also had some issues getting `Not` to work. Maybe I'll just add them as `TestUtils.UnityNull` and `TestUtils.UnityNotNull`.

### CODE REVIEW

**Comment by StephenHodgson** [comment]
This is awesome @Zod-, you rock.

File: Assets/HoloToolkit/Utilities/Scripts/CameraCache.cs [hunk 1]
```diff
@@ -20,6 +20,7 @@ public static Camera Main
         {
             get
             {
+                // ReSharper disable once ConvertIfStatementToReturnStatement
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Utilities/Scripts/CameraCache.cs]: I'm unsure if we should include ReSharper stuff in the project.

Comment (Zod-) [reply Assets/HoloToolkit/Utilities/Scripts/CameraCache.cs]: Ok

**Comment by StephenHodgson** [comment]
One last thing @Zod-, I believe there's a Tests Readme document.  Could you update that document with some of your PR comments that describes it's usage?  

Thanks.

File: Assets/HoloToolkit-UnitTests/Editor/Utilities/UnityNullConstraintTests.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        [Test]
+        public void TestActualNull()
+        {
+            Assert.That(null, Is.UnityNull);
+        }
+
+        [Test]
+        public void TestNullCheckFailWithNormalObjects()
+        {
+            Assert.That(new object(), Is.UnityNull);
```
Comment (Zod-) [inline Assets/HoloToolkit-UnitTests/Editor/Utilities/UnityNullConstraintTests.cs]: I've made a mistake here and this test should not assert true.

**Pull request merged**